pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial.md

```diff
@@ -0,0 +1,262 @@
+
+
+```js
+function initCheckPassword(element) {
+
+};
+```
+
+
+We’ll then see if the element we pass as our argument can pass the defined checks.
+
+
+```js
+function initCheckPassword(element) {
+ const checkPassword = () => {
+    passwordChecks.forEach((item) => {
+      let passed = item.test(element.value);
+       // passed/fail logic 
+    });
+};
+```
+
+User actions can trigger [mutations within the `amp-script` component ]({{g.doc('/content/amp-dev/documentation/components/amp-script.md', locale=doc.locale).url.path}}?format=websites#mutations-and-user-actions), allowing `amp-script` to register [event listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventListener). 
+
+Our function will listen for two events, [`keyup`](https://developer.mozilla.org/en-US/docs/Web/API/Document/keyup_event) for when a user types into the input box, and [`change`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) in case our user pastes their password.
+
+
+```js
+function initCheckPassword(element) {
+ const checkPassword = () => {
+    passwordChecks.forEach((item) => {
+      let passed = item.test(element.value);
+       // passed/fail logic 
+    });
+    // is called when user types in input
+  element.addEventListener("keyup", checkPassword);
+  // is called if user pastes into input 
+  element.addEventListener("change", checkPassword); 
+  };
+};    
+```
+
+
+Inside the `checkPassword` function, we’ll toggle a class that changes our text to green if the check is passed.
+
+```js
+function initCheckPassword(element) {
+  const checkPassword = () => {
+    passwordChecks.forEach((item) => {
+      let passed = item.test(element.value);
+      // captures element
+      let checkText = document.getElementById(item.elementId)
+       // passed/fail logic 
+       checkText.classList.toggle('pass', passed)
+    });
+  } 
+    // is called when user types in input
+    element.addEventListener("keyup", checkPassword);
+    // is called if user pastes into input 
+    element.addEventListener("change", checkPassword);
+};
+```
+
+For this to work, we will need to add a `check` class to all the items and define both `check` and `pass` within the `<style amp-custom>` tag in the page head. 
+
+
+```html
+<ul>
+  <li id="lowercase" class="check">Lowercase letter</li>
+  <li id="capital" class="check">Capital letter</li>
+  <li id="number" class="check">Number</li>
+  <li id="special" class="check">Special character (@$!%*?&)</li>
+  <li id="eight" class="check">At least 8 characters long</li>
+</ul> 
+```
+
+
+
+```css
+.check {
+  color:#c11136;
+}
+.check.pass {
+  color: #2d7b1f;
+} 
+```
+
+
+Refresh the page and type into the password input. The elements corresponding the check passed or failed should turn red or green accordingly!
```

Comment:
"The elements corresponding the check passed or failed" -> "The elements corresponding to whether the check passed or failed"